### ISSUE TITLE
[Feature] <title>modeling_chatglm.py代码疑问

### ISSUE THREAD

**Issue by OuYanXi** [THUDM/ChatGLM-6B#136]
### Is your feature request related to a problem? Please describe.

    def get_position_ids(self, seq, mask_position, device, gmask=False):
        context_length = seq.index(150004) + 1 
        if self.position_encoding_2d:
            seq_length = seq.index(150004)
            position_ids = torch.arange(context_length, dtype=torch.long, device=device)
            if not gmask:
                position_ids[seq_length:] = mask_position
            block_position_ids = torch.cat((
                torch.zeros(seq_length, dtype=torch.long, device=device),
                torch.arange(context_length - seq_length, dtype=torch.long, device=device) + 1
            ))
            position_ids = torch.stack((position_ids, block_position_ids), dim=0)
        else:
            position_ids = torch.arange(context_length, dtype=torch.long, device=device)
            if not gmask:
                position_ids[context_length - 1:] = mask_position

        position_ids = position_ids.unsqueeze(0)

        return position_ids

### Solutions

context_length = seq.index(150004) + 1 
以上代码不是很理解，是不是该改成 context_length = len(seq)，如果不是能帮忙解释一下吗？

**Comment by koth** [comment]
seq可能padding了，index返回最后的一个词的位置

**Comment by duzx16** [comment]
效果是一样的，因为这个函数只会在第一次forward的时候调用，而第一次forward的时候seq的最后一个位置总是150004

**Comment by songyingxin** [comment]
> 效果是一样的，因为这个函数只会在第一次forward的时候调用，而第一次forward的时候seq的最后一个位置总是150004

并不一样，这取决于输入时候的input_ids的构建，我不知道为什么，当我采用padding 的时候，150004 有时候并不在最后一个位置，最后的位置属于 pad_token_id。
